src/java.base/share/classes/java/lang/CharacterDataLatin1.java.template

```diff
@@ -90,6 +90,10 @@ class CharacterDataLatin1 extends CharacterData {
         return (getPropertiesEx(ch) & $$maskOtherLowercase) != 0;
     }
 
+    boolean hasUpperCaseMapping(int ch) {
```

Comment:
This method is incorrectly named; it actually tests that a char has no upper case mapping. I recommend fixing this by keeping the method name while removing the double `!` at this return and at the if down below to simplify the logic.

Reply:
Now use a local variable notUpperCaseEx, I prefer this, without adding a method.

Reply:
I'd have preferred @liach's solution (remove the double negative and keeping the method with name unchanged) but this is OK too.